### PR TITLE
Configure jobs deploying & publishing contracts

### PR DESCRIPTION
**Pull request by michalinacienciala**
We're adding jobs which will deploy and publish Threshold contracts.
`NPM` workflow will build & publish contracts every time there is a
change in contracts or in the files describing package dependencies.
The `-dev` suffix will be used in the name of the published package.
`Solidity` workflow will deploy and publish contracts every time
worklflow is triggered by the `workflow_dispatch` event. Published
package name will point to name of the environment on which the
contracts were deployed.

Note: we're temporarily switched off publishing to NPM registry using
`--dry-run` flag. The flag will be removed before merging of the changes
to the `main` branch.

TODO:
- [x] setup GH secrets
- [x] test the workflows
- [x] cleanup of test configuration

Ref:
https://github.com/threshold-network/solidity-contracts/pull/22

### CODE REVIEW

File: .github/workflows/contracts.yaml [hunk 1]
```diff
@@ -27,8 +39,135 @@ jobs:
...
+      - name: Verify contracts on Etherscan
+        env:
+          ETHERSCAN_API_KEY: ${{ secrets.ETHERSCAN_API_KEY }}
+          CHAIN_API_URL: ${{ secrets.ROPSTEN_ETH_HOSTNAME_HTTP }}
+        # run: |
+        #   yarn run hardhat --network ${{ github.event.inputs.environment }} \
+        #     etherscan-verify --license MIT
+        run: |
+          yarn run hardhat --network ropsten \
+            etherscan-verify --license MIT
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yaml:166]: @Kuba, `T` and `VendingMachine` contracts use `GPL-3.0-or-later` licens, which is not supported by etherscan. Do you think I could force `MIT license here (`yarn run hardhat --network ropsten etherscan-verify --license MIT --force-license`)? I verified that it works locally. Without the `--force-license` or `--license MIT --force-license` I was getting errors preventing from verification).

Comment (nkuba) [reply .github/workflows/contracts.yaml:169]: Looks like GPL v3 should be supported by etherscan: https://etherscan.io/contract-license-types.
I'm not sure what value the command is expecting though.

Comment (michalinacienciala) [reply .github/workflows/contracts.yaml]: I have force using `GPL-3.0` license (`yarn run hardhat --network ropsten etherscan-verify --license GPL-3.0 --force-license`), see cfaa9e1.

**Comment by michalinacienciala** [comment]
Workflows positively tested with testing configuration, see here: https://github.com/threshold-network/solidity-contracts/runs/4210925769?check_suite_focus=true.
I removed the testing configuration, PR is ready for review.

**Review by nkuba — APPROVED** [review]

**Pull request merged**